src/basicpy/basicpy.py

```diff
@@ -415,11 +403,11 @@ def fit(
                 B,
                 I_R,
             )
-            logger.info(f"single-step optimization score: {norm_ratio}.")
-            logger.info(f"mean of S: {float(jnp.mean(S))}.")
+            logger.debug(f"single-step optimization score: {norm_ratio}.")
+            logger.debug(f"mean of S: {float(jnp.mean(S))}.")
             self._score = norm_ratio
             if not converged:
-                logger.warning("single-step optimization did not converge.")
+                logger.debug("single-step optimization did not converge.")
```

Comment:
I think this should be a `warning` as it notifies the user that something is wrong.

Reply:
I changed this to debug because every iteration it does not converge we get a message. This does not appear useful as a warning, since it is probably common and not significant that it does not converge on every iteration. What is more significant is that it does not converge at all, and that is contained in a warning below.

I also made this debug instead of info because I tried to limit info to messages that would give a progress update. I figure debug is important to see the details of each step, but info is only important to see progress of each step. I have one info message for each loop.

If you still think this should be changed to info, I will change it.

Reply:
Hmm, I think I got your point. Actually, there are cases in which all the iterations do not converge but the reweighting iteration converges. If we just make it `info` or `debug` we may miss this case. 
How about checking this convergence after the reweighting loop and make a warning if it didn't converge at the last step?